packages/eslint-plugin/src/rules/block-spacing.ts

```diff
@@ -0,0 +1,166 @@
+     * @returns
+     */
+    function checkSpacingInsideBraces(node: TSESTree.TSEnumDeclaration): void {
+      // Gets braces and the first/last token of content.
+      const openBrace = getOpenBrace(node);
+      const closeBrace = sourceCode.getLastToken(node)!;
+      const firstToken = sourceCode.getTokenAfter(openBrace, {
+        includeComments: true,
+      })!;
+      const lastToken = sourceCode.getTokenBefore(closeBrace, {
+        includeComments: true,
+      })!;
+
+      // Skip if the node is invalid or empty.
+      if (
+        openBrace.type !== AST_TOKEN_TYPES.Punctuator ||
+        openBrace.value !== '{' ||
+        closeBrace.type !== AST_TOKEN_TYPES.Punctuator ||
+        closeBrace.value !== '}' ||
+        firstToken === closeBrace
+      ) {
+        return;
+      }
+
+      // Skip line comments for option never
+      if (!always && firstToken.type === AST_TOKEN_TYPES.Line) {
+        return;
+      }
+
+      // Check.
+      if (!isValid(openBrace, firstToken)) {
+        let loc = openBrace.loc;
+
+        if (messageId === 'extra') {
+          loc = {
+            start: openBrace.loc.end,
+            end: firstToken.loc.start,
+          };
+        }
+
+        context.report({
+          node,
+          loc,
+          messageId,
+          data: {
+            location: 'after',
+            token: openBrace.value,
+          },
+          fix(fixer) {
+            if (always) {
+              return fixer.insertTextBefore(firstToken, ' ');
+            }
+
+            return fixer.removeRange([openBrace.range[1], firstToken.range[0]]);
+          },
+        });
+      }
+      if (!isValid(lastToken, closeBrace)) {
+        let loc = closeBrace.loc;
+
+        if (messageId === 'extra') {
+          loc = {
+            start: lastToken.loc.end,
+            end: closeBrace.loc.start,
+          };
+        }
+        context.report({
+          node,
+          loc,
+          messageId,
+          data: {
+            location: 'before',
+            token: closeBrace.value,
+          },
+          fix(fixer) {
+            if (always) {
+              return fixer.insertTextAfter(lastToken, ' ');
+            }
+
+            return fixer.removeRange([lastToken.range[1], closeBrace.range[0]]);
+          },
+        });
+      }
+    }
+    return {
+      ...baseRules,
+
+      // This code worked "out of the box" for interface and type literal
+      // Enums were very close to match as well, the only reason they it not is that was that enums don't have a body node in the parser
+      // So the opening brace punctuator starts in the middle of the node
+      // `getFirstToken` in the base rule did not filter for the first opening brace punctuator
+      TSInterfaceBody: baseRules.BlockStatement as never,
+      TSTypeLiteral: baseRules.BlockStatement as never,
+      TSEnumDeclaration: checkSpacingInsideBraces,
```

Comment:
Kept the base rule for interface and type literal, though changing to `checkSpacingInsideBraces` will work as well